packages/astro/src/runtime/client/dev-toolbar/apps/audit/index.ts

```diff
@@ -69,8 +70,46 @@ export default {
 
 		await lint();
 
-		document.addEventListener('astro:after-swap', async () => lint());
-		document.addEventListener('astro:page-load', async () => refreshLintPositions);
+		let mutationDebounce: ReturnType<typeof setTimeout>;
+		const observer = new MutationObserver(() => {
+			if (mutationDebounce) {
+				clearTimeout(mutationDebounce);
+			}
+			mutationDebounce = setTimeout(() => {
+				settings.logger.verboseLog('Rerunning audit lints because the DOM has been updated.');
+
+				if ('requestIdleCallback' in window) {
+					window.requestIdleCallback(
+						async () => {
+							lint();
+						},
+						{ timeout: 300 }
+					);
+				} else {
+					// Fallback for old versions of Safari
+					setTimeout(() => {
+						lint();
+					}, 150);
+				}
```

Comment:
Why doe we have two different timeouts?

Reply:
We only want to rerun the lints after the final page mutation, so we debounce it with the first `setTimeout`, but we also want to audit only if the browser is idle (in order to avoid affecting the page's performance). Alas, Safari doesn't support `requestIdleCallback` so we have to "fake it" (we assume that after 150ms, the page is hopefully idle)